src/hooks/useUser.ts

```diff
@@ -19,34 +19,45 @@ const emptyUser = {
 };
 
 async function getUser(): Promise<User> {
-  const res = await window.gapi.client.people.people.get({
-    resourceName: 'people/me',
-    personFields: 'names,emailAddresses,photos',
-  });
-
-  return {
-    name: res.result.names?.[0].displayName as string,
-    email: res.result.emailAddresses?.[0].value as string,
-    image: res.result.photos?.[0].url as string,
-    isLoggedIn: true,
-  };
+  try {
+    const res = await window.gapi.client.people.people.get({
+      resourceName: 'people/me',
+      personFields: 'names,emailAddresses,photos',
+    });
+
+    return {
+      name: res.result.names?.[0].displayName as string,
+      email: res.result.emailAddresses?.[0].value as string,
+      image: res.result.photos?.[0].url as string,
+      isLoggedIn: true,
+    };
+  } catch {
+    localStorage.setItem('accessToken', '');
+    return emptyUser;
+  }
 }
 
-export default function useUser(): { user: User } {
+export default function useUser(): { user: User, mutate: Function } {
   const router = useRouter();
   const [isGapiLoaded] = useGapiClient();
-  const { data: user, error } = useSWR<User>(isGapiLoaded ? '/api/user' : null, getUser);
+  const { data: user, mutate, isLoading } = useSWRImmutable<User>(
+    isGapiLoaded ? '/api/user' : null,
+    getUser,
+    {
+      refreshInterval: 100000,
+    },
+  );
 
   React.useEffect(() => {
-    if (error) {
+    if (!isLoading && user && !user.isLoggedIn) {
```

Comment:
we never have an error now as if there's an error in the fetcher we will return an empty user.

The isLoading part is so we only re-redirect once we have a result.